lib/IRGen/GenMeta.cpp

```diff
@@ -2854,23 +2854,77 @@ static void emitInitializeFieldOffsetVector(IRGenFunction &IGF,
   IGF.Builder.CreateLifetimeEnd(fields, IGM.getPointerSize() * numFields);
 }
 
+static void emitInitializeFieldOffsetVectorWithLayoutString(
+    IRGenFunction &IGF, SILType T, llvm::Value *metadata, bool isVWTMutable) {
+  auto &IGM = IGF.IGM;
+
+  auto *target = T.getStructOrBoundGenericStruct();
+
+  llvm::Value *fieldVector =
+      emitAddressOfFieldOffsetVector(IGF, metadata, target).getAddress();
+
+  // Collect the stored properties of the type.
+  unsigned numFields = getNumFields(target);
+
+  // Ask the runtime to lay out the struct or class.
+  auto numFieldsV = IGM.getSize(Size(numFields));
+
+  // Fill out an array with the field type metadata records.
+  Address fields =
+      IGF.createAlloca(llvm::ArrayType::get(IGM.TypeMetadataPtrTy, numFields),
+                       IGM.getPointerAlignment(), "fieldsMetadata");
+  IGF.Builder.CreateLifetimeStart(fields, IGM.getPointerSize() * numFields);
+  fields = IGF.Builder.CreateStructGEP(fields, 0, Size(0));
+
+  unsigned index = 0;
+  forEachField(IGM, target, [&](Field field) {
+    assert(field.isConcrete() &&
+           "initializing offset vector for type with missing member?");
+    SILType propTy = field.getType(IGM, T);
+    llvm::Value *fieldMetatype = IGF.emitTypeMetadataRef(propTy.getASTType());
```

Comment:
This seems to be incorrect.

Reply:
Have you tested this with recursive types like:
```
public class Cons<T> {
    public var cdr: Cons<T>?
    public var car: T
    public init(_ t: T) {
        car = t
    }
}
```